pandas/io/formats/excel.py

```diff
@@ -385,7 +385,7 @@ def __init__(
     ):
         self.rowcounter = 0
         self.na_rep = na_rep
-        if hasattr(df, "render"):
+        if hasattr(df, "render") and not isinstance(df, DataFrame):
```

Comment:
Can this instead be replaced with an isinstance check for Styler?

Reply:
That was my first idea too, but you get jinja errors, when the tests are executed via the pipelines. 

```
Traceback:
pandas/tests/io/excel/test_style.py:8: in <module>
    from pandas.io.formats.excel import ExcelFormatter
pandas/io/formats/excel.py:26: in <module>
    from pandas.io.formats.style import Styler
pandas/io/formats/style.py:41: in <module>
    jinja2 = import_optional_dependency("jinja2", extra="DataFrame.style requires jinja2.")
pandas/compat/_optional.py:92: in import_optional_dependency
    raise ImportError(msg) from None
E   ImportError: Missing optional dependency 'jinja2'. DataFrame.style requires jinja2. Use pip or conda to install jinja2.
__________ ERROR collecting pandas/tests/io/formats/test_to_excel.py ___________
ImportError while importing test module '/home/vsts/work/1/s/pandas/tests/io/formats/test_to_excel.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
pandas/tests/io/formats/test_to_excel.py:11: in <module>
    from pandas.io.formats.excel import CSSToExcelConverter
pandas/io/formats/excel.py:26: in <module>
    from pandas.io.formats.style import Styler
pandas/io/formats/style.py:41: in <module>
    jinja2 = import_optional_dependency("jinja2", extra="DataFrame.style requires jinja2.")
pandas/compat/_optional.py:92: in import_optional_dependency
    raise ImportError(msg) from None
E   ImportError: Missing optional dependency 'jinja2'. DataFrame.style requires jinja2. Use pip or conda to install jinja2.
```

I changed the comparison to a `DataFrame` afterwards. I saw other tests where this was avoided through an importskip, but the import was done in the test file there, not in the actual code. Should I try changing that back nevertheless?

Reply:
might be easier *just* to check if this is an instance of ``DataFrame`` only.

Reply:
I had the same idea, but I was not sure if it is possible that we get a `Styler` without the render attribute. I figured that this is not an option, but was not 100% sure